src/CodeGen_LLVM.cpp

```diff
@@ -436,9 +432,7 @@ llvm::Module *CodeGen_LLVM::compile(const Module &input) {
     CodeGen_LLVM::optimize_module();
 
     // Disown the module and return it.
-    llvm::Module *m = module;
-    module = NULL;
-    return m;
+    return std::move(module);
```

Comment:
This is returning a field of the class and the "try treating it as an r-value first" thing doesn't apply to members of a class. I'll see if I can remove some of the others. (I put them in as I was getting compile errors here. I just went and read the spec. I know more than I did before but I do not feel enlightened.) Another approach would be to not make the module a member of this class, but I'm not sure that is right either.